authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java

```diff
@@ -0,0 +1,179 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.authorization.ranger;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import com.google.errorprone.annotations.FormatMethod;
+import com.google.errorprone.annotations.FormatString;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.authorization.Privilege;
+import org.apache.gravitino.authorization.SecurableObjects;
+import org.apache.gravitino.connector.authorization.AuthorizationPlugin;
+import org.apache.gravitino.exceptions.AuthorizationHookException;
+import org.apache.ranger.RangerServiceException;
+import org.apache.ranger.plugin.model.RangerPolicy;
+import org.apache.ranger.plugin.util.SearchFilter;
+
+/** Ranger authorization operations hooks interfaces. */
+public abstract class RangerAuthorizationPlugin implements AuthorizationPlugin {
+  protected String catalogProvider;
+  protected RangerClientExt rangerClient;
+  protected String rangerServiceName;
+  /** Mapping Gravitino privilege name to the underlying authorization system privileges. */
+  protected Map<Privilege.Name, Set<String>> mapPrivileges = null;
+
+  /**
+   * Because UserGroupAuthorizationPlugin::onOwnerSet() interface function didn't have param Role,
+   * So we need predefine a role name
+   */
+  public static final String OWNER_ROLE_NAME = "OWNER";
+
+  public static final String MANAGED_BY_GRAVITINO = "MANAGED_BY_GRAVITINO";
+  public static final String POLICY_ITEM_OWNER_USER = "{OWNER}";
+
+  public RangerAuthorizationPlugin() {
+    initMapPrivileges();
+  }
+
+  /**
+   * Different underlying permission system may have different privilege names, this function is
+   * used to initialize the privilege mapping.
+   */
+  protected abstract void initMapPrivileges();
+
+  /**
+   * Translate the privilege name to the corresponding privilege name in the underlying permission
+   */
+  public Set<String> translatePrivilege(Privilege.Name name) {
+    return mapPrivileges.get(name);
+  }
+
+  /** Whether this privilege is underlying permission system supported */
+  protected boolean checkPrivilege(Privilege.Name name) {
+    return mapPrivileges.containsKey(name);
+  }
+
+  @FormatMethod
+  protected static void check(boolean condition, @FormatString String message, Object... args) {
+    if (!condition) {
+      throw new AuthorizationHookException(message, args);
+    }
+  }
+
+  @VisibleForTesting
+  public String formatPolicyName(String roleName, String securableObjectFullName) {
+    return roleName + "-" + securableObjectFullName;
```

Comment:
Why does policyName need to include role information? I think it's enough to use `securableObjectFullName`